### DISCUSSION TITLE
RemBG usage with SAM

### DISCUSSION THREAD

**Discussion by Dpetrovic93** [danielgatis/rembg#538] [Q&A]
Any clarification how to use remove with SAM model. Tnx in advance

**Comment by Franion03** [comment]
I would like to know how to use it as well i am just trying this : 
`rembg p -m sam -x '{ "sam_prompt": [{"type": "rectangle", "data": [300, 400], "label": 1}] }' ./ ./sam`

and it is showing me this error : 
```
{}' is not of type 'array'

Failed validating 'type' in schema:
    {'items': {'properties': {'data': {'items': {'type': 'number'},
                                       'type': 'array'},
                              'label': {'type': 'integer'},
                              'type': {'type': 'string'}},
               'type': 'object'},
     'type': 'array'}

On instance:
    '{}'
100%|████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:00<00:00, 14.94it/s]

```

It is working well with the unet but i can't make it works with sam

**Reply by nonno-cicala** [reply]
I had the same problem. I didn't check the code but it works for me if I use type "point" and I set the two numbers in data array as a point inside the object I want to select. I think "rectangle" works with four numbers in the data array, where the first two represent the top left corner and the last two the bottom right corner of the selection rectangle

**Comment by GiangTMA** [comment]
How can I use Rembg as a library with SAM ?
